fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/rules/AbstractDataAccessRule.java

```diff
@@ -501,23 +501,32 @@ private static List<RelationalExpression> createIntersectionAndCompensation(@Non
      * Private helper method that computes the ordering of the intersection using matches and the common primary key
      * of the data source.
      * @param partitionOrderings partition we would like to intersect
-     * @return a {@link PartialOrder}  of {@link BoundKeyPart} representing a common intersection ordering.
+     * @return a {@link PartialOrder}  of {@link KeyPart}s representing a common intersection ordering.
      */
     @SuppressWarnings("java:S1066")
     @Nonnull
-    private static PartialOrder<BoundKeyPart> intersectionOrdering(@Nonnull final List<List<BoundKeyPart>> partitionOrderings) {
+    private static PartialOrder<KeyPart> intersectionOrderingPartialOrder(@Nonnull final List<List<BoundKeyPart>> partitionOrderings) {
 
         final var orderingPartialOrders =
                 partitionOrderings.stream()
                         .map(boundKeyParts -> {
-                            final var orderingKeyParts =
+                            final var boundKeyPartsPartitions =
                                     boundKeyParts.stream()
+                                            .collect(Collectors.partitioningBy(boundOrderingKey -> boundOrderingKey.getComparisonRangeType() == ComparisonRange.Type.EQUALITY));
+
+                            final var equalityBoundKeyParts = boundKeyPartsPartitions.get(true).stream()
+                                    .map(BoundKeyPart::getKeyPart)
+                                    .collect(ImmutableSet.toImmutableSet());
+                            final var orderingKeyParts =
+                                    boundKeyPartsPartitions.get(false)
+                                            .stream()
                                             .filter(boundOrderingKey -> boundOrderingKey.getComparisonRangeType() != ComparisonRange.Type.EQUALITY)
+                                            .map(BoundKeyPart::getKeyPart)
                                             .collect(ImmutableList.toImmutableList());
 
-                            return PartialOrder.<BoundKeyPart>builder()
+                            return PartialOrder.<KeyPart>builder()
                                     .addListWithDependencies(orderingKeyParts)
-                                    .addAll(boundKeyParts)
```

Comment:
It could definitely be that I am missing something, but adding these points to partial order seems useless, because they can never establish a dependency in the resulting merged partial order (they are mere points in left/right Hasse diagrams and will be carried over as-is in the resulting merged partial orderI guess). Is this correct? if so I guess we can optimise the merging algorithm to just _premerge_ these points since they are eligible/least elements.